### PR TITLE
SPECIFYING INPUT FOR MUTACC IMPORT

### PR DESCRIPTION
**Pull request by adrosenbaum**
To import a new case into the database a .bam and .fastq(optional) is
given. Variant data in JSON format. Case data is also given
in JSON.

### CODE REVIEW

File: mutacc/cli/importing.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+from mutacc.utils.hk_call import find_files
+
+@click.command('import')
+@click.option('--bam', multiple = True, type = click.Path(exists = True), help = 'path to .bam file')
+@click.option('--fastq', multiple = True, type = click.Path(exists = True), help = 'path to .fastq file')
+@click.option('--variant', help = 'Variant data as JSON string, or path to .json file')
+@click.option('--case', help = 'Metadata for the case as JSON string or path to .json file')
+@click.pass_context
+def importing(context,bam,fastq,variant,case):
```
Comment (henrikstranneheim) [inline mutacc/cli/importing.py]: How about spaces between the parameters i.e. "context, bam". Makes it easier to read

File: mutacc/cli/importing.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    """
+        Import cases to the database
+    """
+            
+    case = parse_JSON(case)
```
Comment (henrikstranneheim) [inline mutacc/cli/importing.py]: Nice! You might want to consider adding this to a "parse" package and import it to the Cli.

File: mutacc/cli/importing.py (hunk 1 above)
Comment (moonso) [reply mutacc/cli/importing.py]: I agree, that is a standard that we use, should look like:
```python
def importing(context, bam, fastq, variant, case):
```

File: mutacc/cli/importing.py (hunk 2 above)
Comment (moonso) [reply mutacc/cli/importing.py]: convention is to use small letters only in function names.
Classes start with big letter.

File: mutacc/cli/importing.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        click.echo(path)
+    
+    click.echo(case)
+    click.echo(variant)
+
+def parse_JSON(json_format):
+
+    if json_format[0] == '~':
+
+        json_format = os.path.expanduser(json_format)
```
Comment (moonso) [inline mutacc/cli/importing.py]: I hope you are using python version >=3.6. In that case you might want to have a look at the [path lib](https://docs.python.org/3/library/pathlib.html)

File: mutacc/cli/importing.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    else:
+
+        try: 
+
+            json_format = json.loads(json_format)
+
+        except json.JSONDecodeError:
+                    
+            sys.exit("Not valid Json")
```
Comment (moonso) [inline mutacc/cli/importing.py]: You should use `raise Exception` here and catch it wherever the function is run. https://docs.python.org/3/tutorial/errors.html

File: mutacc/cli/root.py [hunk 5]
```diff
@@ -3,7 +3,7 @@
 import click
 import coloredlogs
 
-from . import export, add, delete, init
+from . import export, importing, delete, init
```
Comment (moonso) [inline mutacc/cli/root.py]: Suggestion is to use something like
```python
from .export import export as export_command
```
To not mess up the name space

**Pull request merged**